packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx

```diff
@@ -0,0 +1,627 @@
+        onMouseOutAction: () => {
+          this.setState({ hoveredLegend: '' });
+        },
+      };
+    });
+
+    return <Legends legends={legends} centerLegends {...this.props.legendProps} />;
+  };
+
+  /**
+   * This function checks if the given legend is highlighted or not.
+   * A legend can be highlighted in 2 ways:
+   * 1. selection: if the user clicks on it
+   * 2. hovering: if there is no selected legend and the user hovers over it
+   */
+  private _legendHighlighted = (legend: string) => {
+    return (
+      this.state.selectedLegend === legend || (this.state.selectedLegend === '' && this.state.hoveredLegend === legend)
+    );
+  };
+
+  /**
+   * This function checks if none of the legends is selected or hovered.
+   */
+  private _noLegendHighlighted = () => {
+    return this.state.selectedLegend === '' && this.state.hoveredLegend === '';
+  };
+
+  private _handleFocus = (focusEvent: React.FocusEvent<SVGElement>, focusedElement: string) => {
+    const hoverXValue: string =
+      'Current value is ' +
+      (this.props.chartValueFormat === GaugeValueFormat.Fraction
+        ? `${((this._sweepFraction[0] / this._sweepFraction[1]) * 100).toFixed()}%`
+        : `${this._sweepFraction[0]}/${this._sweepFraction[1]}`);
+    let total = this._minValue - 1;
+    const hoverYValues: IYValue[] = this._segments.map(segment => {
+      const yValue: IYValue = {
+        legend: segment.legend,
+        y: `${total + 1} - ${total + segment.size}`,
+        color: segment.color,
+      };
+      total += segment.size;
+      return yValue;
+    });
+    this.setState({
+      focusedElement,
+      calloutTarget: focusEvent.target,
+      isCalloutVisible: true,
+      hoverXValue,
+      hoverYValues,
+    });
+  };
+
+  private _handleBlur = () => {
+    this.setState({
+      focusedElement: '',
+      calloutTarget: null,
+      isCalloutVisible: false,
+      hoverXValue: '',
+      hoverYValues: [],
+    });
+  };
+
+  private _handleCalloutDismiss = () => {
+    this.setState({ isCalloutVisible: false });
+  };
+
+  private _handleMouseOver = (mouseEvent: React.MouseEvent<SVGElement>, hoveredElement: string) => {
+    if (this._calloutAnchor === hoveredElement) {
+      return;
+    }
+
+    this._calloutAnchor = hoveredElement;
+    const hoverXValue: string =
+      'Current value is ' +
+      (this.props.chartValueFormat === GaugeValueFormat.Fraction
+        ? `${((this._sweepFraction[0] / this._sweepFraction[1]) * 100).toFixed()}%`
+        : `${this._sweepFraction[0]}/${this._sweepFraction[1]}`);
+    let total = this._minValue - 1;
+    const hoverYValues: IYValue[] = this._segments.map(segment => {
+      const yValue: IYValue = {
+        legend: segment.legend,
+        y: `${total + 1} - ${total + segment.size}`,
+        color: segment.color,
+      };
+      total += segment.size;
+      return yValue;
+    });
+    this.setState({
+      calloutTarget: mouseEvent.target,
+      isCalloutVisible: true,
```

Comment:
HandleFocus and HandleMouseOver have almost same code. Can we avoid code duplication

Reply:
fixed